src/main/java/com/pngencoder/PngEncoder.java

```diff
@@ -28,29 +30,47 @@ public class PngEncoder {
     private final PngEncoderSrgbRenderingIntent srgbRenderingIntent;
     private final PngEncoderPhysicalPixelDimensions physicalPixelDimensions;
 
-    private PngEncoder(
-            BufferedImage bufferedImage,
-            int compressionLevel,
-            boolean multiThreadedCompressionEnabled,
+    private final PredictorEncoding predictorEncoding;
+
+    private enum PredictorEncoding {
+        DEFAULT {
+            @Override
+            boolean isEnabled(int compressionLevel, boolean multiThreadedCompressionEnabled) {
+                return compressionLevel >= 6 && !multiThreadedCompressionEnabled;
```

Comment:
I can't make up my mind. I'm on the fence between this "sane" default or whether "as fast as possible" should be the default.

Reply:
I guess we can still claim to be "very fast" because this feature can simply be turned off for very intense applications.

Reply:
I'm indifferent here. I don't have a problem with making this a simple boolean with false being the default.

Reply:
BTW, why would the predictor be disabled by default when running multi-threaded?

Reply:
I made a separate branch just to wrap my head around this. I refactored the (now) three compression arguments into a class, moving validation and defaults to there.
264de04

Reply:
I can't seem to create a reverse Pull Request into your branch. I'd probably have to create a new fork off of your branch and make a Pull Request from there into your branch. meh.

Reply:
Beside this being a API breaking change, it looks good.

I planing to contribute my indexed PNG encoder I built some years ago (after this changes are merged). That would be another compression setting. (i.e. tryWithIndexEncoding - as indexed only allows 256 colors this can and will fail for many images). So putting it into its one class makes sense.

Should I just try to pull your changes into this branch?

Reply:
Try it!
1b85b9f

Reply:
I actually like that we would break the API. The users should consider whether they want to use the new predictor or not. Since it might be turned on by default on higher compression levels (which is also nice).

Reply:
I did a comment on the commit. After thinking a little bit about it I don't like an extra compression setting class.

To not surprise existing users we can also just default to not use the predictor by default. 

And give some examples in the README.md how to configure for maximum speed, minimal file size etc.

Reply:
https://github.com/pngencoder/pngencoder/issues/26